scripts/generated-files/lint-rules-docs.ts

```diff
@@ -176,35 +176,70 @@ export async function main() {
 
 				let hasInvalid = false;
 				let hasValid = false;
-				for (const {invalid, valid} of cases) {
-					if (invalid && invalid.length > 0) {
-						hasInvalid = true;
-					}
-					if (valid && valid.length > 0) {
-						hasValid = true;
+				for (const singleCase of cases) {
+					if (Array.isArray(singleCase)) {
+						for (const {invalid, valid} of singleCase) {
+							if (invalid && invalid.length > 0) {
+								hasInvalid = true;
+							}
+							if (valid && valid.length > 0) {
+								hasValid = true;
+							}
+						}
+					} else {
+						const {invalid, valid} = singleCase;
+						if (invalid && invalid.length > 0) {
+							hasInvalid = true;
+						}
+						if (valid && valid.length > 0) {
+							hasValid = true;
+						}
 					}
 				}
```

Comment:
I think this would be a bit more DRY

```suggestion
				for (const elem of cases) {
					const cases = Array.isArray(elem) ? elem : [elem];
					for (const {invalid, valid} of cases) {
						if (invalid && invalid.length > 0) {
							hasInvalid = true;
						}
						if (valid && valid.length > 0) {
							hasValid = true;
						}
					}
				}
```